weave-js/src/common/components/FileBrowser.tsx

```diff
@@ -608,103 +608,94 @@ interface CodePreviewProps {
-        .join('\n');
+        }
 
-      setDataVal(truncated);
-    },
-    [setDataVal, file.name]
-  );
-  const setError = useCallback(
-    (errorString?: string) => setErrorVal(errorString || 'Error loading file'),
-    [setErrorVal]
-  );
+        // Truncate long lines
+        const truncated = lines
+          .map(line => {
+            if (line.length > 1000) {
+              return (
+                line.slice(0, 1000) + ' (line truncated to 1000 characters)'
+              );
+            } else {
+              return line;
+            }
+          })
+          .join('\n');
+
+        setDataVal(truncated);
+      },
+      [setDataVal, file.name]
+    );
+    const setError = useCallback(
+      (errorString?: string) =>
+        setErrorVal(errorString || 'Error loading file'),
+      [setErrorVal]
+    );
 
-  // We don't pass a fallback to allow dev mode zero byte files to render
-  const loading = useLoadFile(file, {
-    onSuccess: setData,
-    onFailure: setError,
-  });
-  useEffect(() => {
-    if (ref.current != null) {
-      Prism.highlightElement(ref.current);
+    // We don't pass a fallback to allow dev mode zero byte files to render
+    const loading = useLoadFile(file, {
+      onSuccess: setData,
+      onFailure: setError,
+    });
+    useEffect(() => {
+      if (ref.current != null && file.sizeBytes / 1024 < 1024) {
+        Prism.highlightElement(ref.current);
+      }
+    });
+    if (error != null) {
+      return <Segment textAlign="center">{error}</Segment>;
     }
-  });
-  if (error != null) {
-    return <Segment textAlign="center">{error}</Segment>;
-  }
-  if (loading) {
-    return <Loader name="code-preview-loader" />;
-  }
-  // HACKING TO DISPLAY VOC
-  // if (file.name.endsWith('.xml')) {
-  //   const parser = new DOMParser();
-  //   const xmlDoc = parser.parseFromString(data, 'text/xml');
-  //   const anno = xmlDoc.getElementsByTagName('annotation')[0];
-  //   if (anno != null) {
-  //     for (let i = 0; i < anno.childNodes.length; i++) {
-  //       const node = anno.childNodes[i];
-  //       if (node.nodeType !== Node.TEXT_NODE && node.nodeName === 'filename') {
-  //         const filename = node.childNodes[0].textContent;
-  //         console.log('FILE NAME', filename);
-  //         // const imageFile = (node as any).getElementsByTagName('filename')[0];
-  //         // console.log('IMAGE FILE', imageFile);
-  //       }
-  //       console.log(node);
-  //     }
-  //     // anno.childNodes[]
-  //     // console.log('VOC!');
-  //   }
-  // }
-  return (
-    <div
-      style={{
-        background: 'white',
-        border: '1px solid #eee',
-        padding: 16,
-      }}>
-      <pre
+    if (loading) {
+      return <Loader name="code-preview-loader" />;
+    }
+
+    return (
+      <div
         style={{
-          maxWidth: '100%',
+          background: 'white',
+          border: '1px solid #eee',
+          padding: 16,
+          height: 'calc(100vh - 150px)',
```

Comment:
the page craps out when we use the normal scroll bar in large files so having this preview scroll on its own prevents that

Reply:
This might be better as an in-code comment